### PR TITLE
UI error states

### LINKED ISSUE #174: Missing data and 404 treatment
**Is your feature request related to a problem? Please describe.**
- When data is unavailable, the page fails to load anything at all. 
- When we hit a 404 error, we don't have any particular treatment. 

**Describe the solution you'd like**
- Data for a specific chart: Show an error-message that is restricted to that particular chart. 
- Data for a whole page: Show an error message in lieu of the chart sections. 
- 404 Page: Well-designed and informative page that encourages the user to navigate back to the [ND DOCR homepage](http://docr.nd.gov) and/or [Recidiviz homepage](http://www.recidiviz.org)

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Adds error handling and UI for three error states for which the site currently presents a broken or otherwise substandard interface: 
- Error panel for errors contained to a single chart
- Error page for errors that are not contained to a single chart (e.g. widespread data loading failure)
 - "404 page" for invalid URLs

Screen Shot 2020-08-27 at 1 56 13 PM
Screen Shot 2020-08-27 at 1 56 38 PM
Screen Shot 2020-08-27 at 1 57 05 PM

I used the Sentry React SDK to implement this to ensure that these errors are still reported to Sentry when we catch them (along with a tag that indicates how much of the page was affected). This does not, however, log invalid URL visits to Sentry, because there isn't actually any error to catch ... I'm not so sure we would really want that in Sentry anyway?

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #174 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/page-not-found/PageNotFound.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+`;
+
+export default function PageNotFound() {
+  return (
+    <Wrapper>
+      <HeadingTitle>Error 404: Page not found.</HeadingTitle>
+      <HeadingDescription>
+        <p>The page that you are looking for is missing.</p>
+        <p>
+          Go to <a href="https://docr.nd.gov">docr.nd.gov</a>
```
Comment (jessex) [inline src/page-not-found/PageNotFound.js:17]: Do we really want to redirect all the way back to the DOCR and not the root of the app?

File: src/site-layout/SiteLayout.js [hunk 2]
```diff
@@ -24,7 +26,7 @@ const SiteContainer = styled.div`
 
 const BodyWrapper = styled(PageWidthContainer)`
   margin: 0 auto;
-  min-height: 100vh;
+  min-height: calc(100vh - ${(props) => props.theme.footerHeight});
```
Comment (jessex) [inline src/site-layout/SiteLayout.js:29]: This doesn't have any impact on your previous IE11 footer fix, right?

File: src/page-not-found/PageNotFound.js (hunk 1 above)
Comment (macfarlandian) [reply src/page-not-found/PageNotFound.js:17]: I will defer to @hobuobi on this one, I wasn't really part of the content conversations for these messages

**Comment by macfarlandian** [comment]
@jessex I verified that events were being sent to the Sentry API via my browser network inspector, and that Sentry was receiving them. Couldn't see the full contents of the error logs in Sentry since we are currently filtering out logs from `localhost` but if you want to make extra double sure you could either turn off that filter for a bit or elevate my permissions so I can do it myself?

File: src/site-layout/SiteLayout.js (hunk 2 above)
Comment (macfarlandian) [reply src/site-layout/SiteLayout.js:29]: I can confirm that it does not!

**Comment by jessex** [comment]
@macfarlandian I _believe_ I just granted you the necessary permissions to disable that filtering. Give it a shot and let me know?

**Comment by macfarlandian** [comment]
👍 I turned off the filter, verified the errors show up in Sentry, and then turned it back on to prevent noise

File: src/page-not-found/PageNotFound.js (hunk 1 above)
Comment (hobuobi) [reply src/page-not-found/PageNotFound.js:17]: Sorry about the late response here! 
We should definitely refer to the root of the app in addition to this (probably as the first option). I'd suggest the following: 

> Click [here](http://dashboard.docr.nd.gov) to return to the homepage of the dashboard, or visit the North Dakota Department of Corrections at [docr.nd.gov](http://docr.nd.gov).

**Pull request merged**